wire-ios-data-model/Source/Model/UserClient/UserClient.swift

```diff
@@ -503,37 +500,98 @@ public extension UserClient {
         syncMOC.performGroupedBlock({ [unowned syncMOC] () -> Void in
             guard let obj = try? syncMOC.existingObject(with: selfObjectID),
                   let syncClient = obj as? UserClient,
-                  let sessionIdentifier = syncClient.sessionIdentifier,
                   let syncSelfClient = ZMUser.selfUser(in: syncMOC).selfClient()
             else { return }
 
             if syncSelfClient == syncClient {
-                // TODO: [John] use flag here
-                syncSelfClient.keysStore.encryptionContext.perform({ (sessionsDirectory) in
-                    syncClient.fingerprint = sessionsDirectory.localFingerprint
-                    syncMOC.saveOrRollback()
-                })
-            }
-            else {
+                syncClient.fingerprint = syncClient.localFingerprint
+                syncMOC.saveOrRollback()
+            } else {
                 if !syncClient.hasSessionWithSelfClient {
                     syncSelfClient.missesClient(syncClient)
                     syncSelfClient.setLocallyModifiedKeys(Set(arrayLiteral: ZMUserClientMissingKey))
                     syncMOC.saveOrRollback()
                 }
                 else {
-                    // TODO: [John] use flag here
-                    syncSelfClient.keysStore.encryptionContext.perform({ (sessionsDirectory) in
-                        syncClient.fingerprint = sessionsDirectory.fingerprint(for: sessionIdentifier)
-                        if syncClient.fingerprint == nil {
-                            zmLog.error("Cannot fetch fingerprint for client \(syncClient.sessionIdentifier!)")
-                        } else {
-                            syncMOC.saveOrRollback()
-                        }
-                    })
+                    syncClient.fingerprint = syncClient.remoteFingerprint
+                    guard syncClient.fingerprint != nil else {
+                        zmLog.error("Cannot fetch fingerprint for client \(syncClient.sessionIdentifier!)")
+                        return
+                    }
+                    syncMOC.saveOrRollback()
                 }
             }
         })
     }
+
+}
+
+// MARK: - Fetch fingerprint
+
+extension UserClient {
+
+    private var remoteFingerprint: Data? {
```

Comment:
I think this should be a method rather than a computed property because it involves a bit of work which is not apparent if it's exposed as a property.

Reply:
Done